example/glue/ArrayletObjectModel.hpp

```diff
@@ -56,6 +56,14 @@ class GC_ArrayletObjectModel
 	{
 		/* No-op */
 	}
+
+#if defined(OMR_ENV_DATA64)
+	MMINLINE void
+	setObjectDataAddr(omrobjectptr_t arrayPtr, void *address)
+	{
+		Assert_MM_unreachable();
```

Comment:
Not quite sure, but we may hit this through compactor path (what might be enabled in OMR/example, unlike scavenge), what would be visible through omr PR testing.

Perhaps Assert_MM_unimplemented() is more appropriate (pure guess, it might just emit a message, and not lead to core dump like what Assert_MM_unreachable() is likely to do)!?

There are ways to run OMR only tests locally, but again, I don't know how to do them.

Reply:
Both of them end up calling `assert(0)` which eventually aborts the application: https://github.com/eclipse/omr/blob/master/gc/base/ModronAssertions.h#L62::L72
> assert() prints an error message to standard error and terminates the program by calling abort(3).

Reply:
I guess then, either Compactor is not even enabled in OMR example or does not happen to trigger in the tests.
I honestly don't know how to tell quickly from config files.

Reply:
Would it be better to remove the assert all together to be safe and just have an empty body?
```
MMINLINE void
	setObjectDataAddr(omrobjectptr_t arrayPtr, void *address)
	{
		/* no-op */
        }
```

Reply:
I believe compactor is disabled afterall, if it's to be believed cmake for OMR (OMR_GC_MODRON_COMPACTION is set to OFF), so I either way is ok. 

However, I'd vote for dropping 'Object' from the name - class/instance name gives context, and clear we are fixing something in (indexable) object and not to some meta structure etc. Up to you, though.